File: Count zeros in sorted matrix (rows and cols sorted descending).py
def count_zeros(matrix):
    n=len(matrix)
    count=0
    i,j=0,n-1
    while i<n and j>=0:
        if matrix[i][j]==0:
            count+=j+1
            i+=1
        else:
            j-=1
    return count

print(count_zeros([[1,1,0],[1,0,0],[0,0,0]]))

File: Count zeros in sorted matrix (rows and cols sorted descending).c
#include <stdio.h>

int countZeros(int n, int mat[n][n]){
    int count=0;
    int i=0, j=n-1;
    while(i<n && j>=0){
        if(mat[i][j]==0){
            count += j+1; 
            i++;
        } else {
            j--;
        }
    }
    return count;
}

int main(){
    int mat[3][3]={{1,1,0},{1,0,0},{0,0,0}};
    printf("Zeros=%d\n",countZeros(3,mat));
    return 0;
}
